spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/MiniBatch.scala

```diff
@@ -95,11 +182,407 @@ class TensorMiniBatch[T](
+        nInputs: Int,
+        nTargets: Int,
+        featurePaddingParam: Option[FeaturePaddingParam[T]] = None,
+        labelPaddingParam: Option[LabelPaddingParam[T]] = None)(
+    implicit ev: TensorNumeric[T]): MiniBatch[T] = {
+    new ArrayTensorMiniBatch[T](Array.tabulate(nInputs)(_ => Tensor[T]()),
+      Array.tabulate(nTargets)(_ => Tensor[T]()),
+      featurePaddingParam, labelPaddingParam)
+  }
+
+  def apply[T: ClassTag](input: Tensor[T], target: Tensor[T]): MiniBatch[T] = {
+    MiniBatch[T](Array(input), Array(target))
+  }
+
+  def apply[T: ClassTag](input: Array[Tensor[T]], target: Tensor[T]): MiniBatch[T] = {
+    MiniBatch[T](input, Array(target))
+  }
+
+  def apply[T: ClassTag](input: Array[Tensor[T]], target: Array[Tensor[T]]): MiniBatch[T] = {
+    new ArrayTensorMiniBatch[T](input, target)
+  }
+
+  def apply[T: ClassTag](input: Tensor[T]): MiniBatch[T] = {
+    MiniBatch[T](Array(input), new Array[Tensor[T]](0))
+  }
+
+  def apply[T: ClassTag](input: Array[Tensor[T]]): MiniBatch[T] = {
+    MiniBatch[T](input, new Array[Tensor[T]](0))
+  }
+
+  private def resizeData[T: ClassTag](
+        data: Array[Tensor[T]],
+        // 1st Seq is batchSize, 2nd Array is number of features, 3th Array is feature size
+        sampleSize: Seq[Array[Array[Int]]],
+        longestData: Option[Array[Int]],
+        paddingStrategy: Option[PaddingStrategy] = None): Unit = {
+    // Size of input data. 1st Array is number of input, 2nd Array is input size.
+    val finalSizes = if (longestData.isDefined) {
+      val longest = longestData.get
+      val sizes = new Array[Array[Int]](longest.length)
+
+      var i = 0
+      while (i < sizes.length) {
+        // Set i-th input's size
+        sizes(i) = Array(sampleSize.length) ++ sampleSize(longest(i))(i)
+        i += 1
+      }
+
+      if (paddingStrategy.isDefined) {
+        paddingStrategy.get.padding(sizes)
+      }
+      sizes
+    } else {
+      val sizes = new Array[Array[Int]](sampleSize(0).length)
+      var i = 0
+      while (i < sizes.length) {
+        // Set i-th input's size
+        sizes(i) = Array(sampleSize.length) ++ sampleSize(0)(i)
+        i += 1
+      }
+      sizes
+
+    }
+
+    // resize
+    var i = 0
+    while (i < finalSizes.length) {
+      data(i).resize(finalSizes(i))
+      i += 1
+    }
+
+  }
+
+  private[bigdl] def arraySampleToMiniBatch[T: ClassTag](
+      samples: Seq[Sample[T]],
+      miniBatch: ArrayTensorMiniBatch[T],
+      longestFeature: Option[Array[Int]] = None,
+      longestLabel: Option[Array[Int]] = None,
+      featurePadding: Option[Array[Tensor[T]]] = None,
+      featurePaddingStrategy: Option[PaddingStrategy] = None,
+      labelPadding: Option[Array[T]] = None,
+      labelPaddingStrategy: Option[PaddingStrategy] = None
+      )(implicit ev: TensorNumeric[T]): MiniBatch[T] = {
+    val inputs = miniBatch.inputData
+    val targets = miniBatch.targetData
+
+    val featureSizes = samples.map(_.getFeatureSize())
```

Comment:
we need to resize when `batchSize != samples.length || featurePaddingParam.isDefined ||
      labelPaddingParam.isDefined`, am I right?